addons/pos_online_payment/models/payment_transaction.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models, tools
+from odoo.exceptions import ValidationError
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    pos_order_id = fields.Many2one('pos.order', string='POS Order', help='The Point of Sale order linked to the payment transaction', readonly=True)
+
+    @api.model
+    def _compute_reference_prefix(self, provider_code, separator, **values):
+        """ Override of payment to compute the reference prefix based on POS-specific values.
+
+        :return: The computed reference prefix if POS order id is found, the one of `super` otherwise
+        :rtype: str
+        """
+        pos_order_id = values.get('pos_order_id')
+        if pos_order_id:
+            pos_order = self.env['pos.order'].sudo().browse(pos_order_id).exists()
+            if pos_order:
+                return pos_order.pos_reference
+        return super()._compute_reference_prefix(provider_code, separator, **values)
+
+    def _set_authorized(self, state_message=None, **kwargs):
+        """ Override of payment to process POS online payments automatically. """
+        super()._set_authorized(state_message=state_message, **kwargs)
```

Comment:
It is a very bad idea to perform complex processing at this point because a server error will trigger a cursor rollback which will undo the creation of the transaction. You'll lose track of the payment, but it will still have been made on the external system.

Let `_reconcile_after_done` do the heavy-lifting instead.

Reply:
Ok thank you for the information. So it is impossible to have a transaction where _set_authorized would be called and _reconcile_after_done would never be called (just to be sure) ?

Reply:
The goal is to detect when a transaction is a successful payment to make the POS order as paid. Good to know _reconcile_after_done is enough to do that.

Reply:
Hello @AntoineVDV, I looked at the code and the sale module also overrides _set_authorized, why would POS would process payments differently than sale ?

Reply:
This kind of override is being removed from all apps as we speak. See https://github.com/odoo/enterprise/commit/126f0087dd8c9dbd9de396a5a14ae05f2a091fba for example.

The override in `sale` will be removed too, but it's far from being as problematic as the one in `pos_online_payment`, which creates `account.payment` records. The latter is the easiest way to make sure some payments will disappear into thin air every now and then.

Reply:
Ok thank you